File: java/src/main/java/winium/elements/desktop/ListBox.java
package winium.elements.desktop;

import org.openqa.selenium.WebElement;
import org.openqa.selenium.By;
import org.openqa.selenium.remote.RemoteWebElement;
import winium.elements.desktop.extensions.ByExtensions;

import java.util.HashMap;

public class ListBox extends DesktopElement {
    static final String SCROLL_TO_LIST_BOX_ITEM = "scrollToListBoxItem";

    public ListBox(WebElement element) {
        super(element);
    }

    public RemoteWebElement scrollTo(By by) {
        HashMap<String, Object> parameters = new HashMap<String, Object>();
        parameters.put("id", this.getId());
        parameters.put("using", ByExtensions.getStrategy(by));
        parameters.put("value", ByExtensions.getValue(by));
        return this.createRemoteWebElementFromResponse(this.execute(SCROLL_TO_LIST_BOX_ITEM, parameters));
    }
}

File: dotnet/src/Winium.Elements.Desktop/ListBox.cs
namespace Winium.Elements.Desktop
{
    #region using

    using System.Collections.Generic;

    using OpenQA.Selenium;
    using OpenQA.Selenium.Remote;

    using Winium.Elements.Desktop.Extensions;

    #endregion

    public class ListBox : DesktopElement
    {
        #region Constants

        private const string ScrollToListBoxItem = "scrollToListBoxItem";

        #endregion

        #region Constructors and Destructors

        public ListBox(IWebElement element)
            : base(element)
        {
        }

        #endregion

        #region Public Methods and Operators

        public RemoteWebElement ScrollTo(By by)
        {
            var response = this.Execute(
                ScrollToListBoxItem,
                new Dictionary<string, object>
                    {
                        { "id", this.Id },
                        { "using", by.GetStrategy() },
                        { "value", by.GetValue() },
                    });

            return this.CreateRemoteWebElementFromResponse(response);
        }

        #endregion
    }
}
